infrastructure/src/main/java/org/corfudb/infrastructure/log/AbstractLogSegment.java

```diff
@@ -0,0 +1,45 @@
+package org.corfudb.infrastructure.log;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.corfudb.protocols.wireprotocol.LogData;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.nio.channels.FileChannel;
+import java.util.Iterator;
+
+/**
+ * TODO: add javadoc.
+ *
+ * Created by WenbinZhu on 5/28/19.
+ */
+@Getter
+@RequiredArgsConstructor
+public abstract class AbstractLogSegment implements AutoCloseable {
+
+    final long startAddress;
+
+    @NonNull
+    final String fileName;
+
+    @NonNull
+    final FileChannel writeChannel;
+
+    @NonNull
+    final FileChannel readChannel;
+
+    // TODO: persist revision in header.
+    // Used for checking equality between stream log and garbage log segment in recovery.
+    long compactRevision;
+
+    // public abstract LogData read(long address);
+
+    public abstract void append(long address, LogData entry);
+
+    // public Iterator<LogData> iterator() {
+    //     throw new NotImplementedException();
+    // }
+
+    public abstract void close();
```

Comment:
Why abstract function?

Reply:
StreamLogSegment and GrabageLogSegment may have different resources to close, so it's an abstract, if later implementation finishes and turns out they are the same, I will change it to non-abstract.